README.md

```diff
@@ -404,9 +404,21 @@ source repository. This is most often the case for nightly-only
 software that pins to a revision from the release archives.
 
 In these cases the toolchain can be named in the project's directory
-in a file called `rust-toolchain`, the content of which is the name of
-a single `rustup` toolchain, and which is suitable to check in to
-source control. This file has to be encoded in US-ASCII (if you are on
+in a file called `rust-toolchain`, the content of which is either the name of
+a single `rustup` toolchain, or a TOML file with the following layout:
+
+``` toml
+[toolchain]
+channel = "nightly-2020-07-10"
+components = [ "rustfmt", "rustc-dev" ]
+targets = [ "wasm32-unknown-unknown", "thumbv2-none-eabi" ]
+```
+
+If the TOML format is used, the `[toolchain]` section and the `channel` entry
```

Comment:
I wonder if we need `channel` to be mandatory -- people may want to use this new more flexible format just to add a target such as `wasm32-unknown-unknown` without stipulating a particular channel.